flink-core/src/main/java/org/apache/flink/api/common/serialization/DeserializationSchema.java

```diff
@@ -42,14 +42,22 @@
 @Public
 public interface DeserializationSchema<T> extends Serializable, ResultTypeQueryable<T> {
 
+	/**
+	 * @deprecated Use {@link #deserialize(ConsumerRecordMetaInfo)} .
```

Comment:
Unnecessary space before period at the end.

Reply:
For the deprecation, I would recommend explaining why the new deserialize method is more superior.

Reply:
I added some more text to the javadoc explaining that implementing the other `deserialize` method has more benefits.